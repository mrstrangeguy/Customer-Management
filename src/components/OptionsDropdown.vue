<template>
  <div class="relative z-100 cursor-pointer w-fit">
    <div
      v-if="!profileURL"
      class="relative overflow-hidden flex py-1.5 z-50 rounded-sm transition-all cursor-pointer pl-3 pr-2 w-fit hover:bg-hover"
      @click="toggleContentVisibility"
      role="button"
    >
      <i
        v-if="icon"
        :class="{
          'default-icon block font-normal block w-4 text-base leading-4 align-middle text-center': true,
          [icon]: true,
        }"
      />
      <span
        id="heading-text"
        class="block text-title leading-4.5 text-black font-medium tracking-title"
        >{{ text }}
      </span>
      <i
        :class="{
          'default-icon ml-1 block w-4 text-base leading-4 text-center align-middle': true,
          [arrowIcon]: true,
        }"
      />
    </div>
    <img
      v-if="profileURL"
      class="block w-7 h-7 object-cover object-top aspect-square border rounded-full"
      :src="profileURL"
      alt="profile-img"
      @click="toggleContentVisibility"
    />
    <div
      :class="{
        'absolute cursor-pointer bg-white transition-all duration-200 border-b w-auto p-px shadow-dropdown1': true,
        'opacity-100 z-150': isHeaderClicked,
        'opacity-0 z-behind': !isHeaderClicked,
        'top-7 left-0': !profileURL,
        'right-0': profileURL,
      }"
    >
      <slot name="dropdown-items" />
    </div>
  </div>
</template>

<script setup lang="ts">
import { ref } from "vue";

//types
type OptionsDropdownProps = {
  text: string;
  arrowIcon: string;
  icon: string;
  profileURL: string;
};

withDefaults(defineProps<OptionsDropdownProps>(), {
  arrowIcon: "",
  icon: "",
  profileURL: "",
  text: "",
});

const isHeaderClicked = ref(false);

const toggleContentVisibility = () => {
  isHeaderClicked.value = !isHeaderClicked.value;
};
</script>

<style lang="scss" scoped></style>

<!-- <options-dropdown
    text="ALL"
    arrow-icon="dx-icon dx-icon-spindown dx-icon-right"
  >
    <template v-slot:dropdown-items>
      <div class="w-[94.97px] bg-[rgba(0,0,0,.12)] mt-1">
        <span
          class="block text-[13px] leading-[16px] px-[11px] pt-[10px] pb-[9px]"
          >All</span
        >
      </div>
      <div class="w-[94.97px] mt-1">
        <span
          class="block text-[13px] leading-[16px] px-[11px] pt-[10px] pb-[9px]"
          >All</span
        >
      </div>
      <div class="w-[94.97px] mt-1">
        <span
          class="block text-[13px] leading-[16px] px-[11px] pt-[10px] pb-[9px]"
          >All</span
        >
      </div>
      <div class="w-[94.97px] mt-1">
        <span
          class="block text-[13px] leading-[16px] px-[11px] pt-[10px] pb-[9px]"
          >All</span
        >
      </div>
    </template>
  </options-dropdown>

  <options-dropdown
    icon="dx-icon dx-icon-export"
    arrow-icon="dx-icon dx-icon-spindown dx-icon-right"
  >
    <template v-slot:dropdown-items>
      <div class="w-[94.97px] mt-1 flex w-[220.53px]">
        <div
          class="flex items-center text-[13px] leading-[16px] px-[11px] pt-[10px] pb-[9px]"
        >
          <div class="w-[29px]">
            <i
              class="default-icon font-medium text-[#0000008a] w-[18px] h-[18px] block dx-icon dx-icon-xlsxfile dx-list-item-icon text-[18px] leading-[18px]"
            ></i>
          </div>
          <div class="leading-[15px]">Export all data to Excel</div>
        </div>
      </div>
    </template>
  </options-dropdown> -->
