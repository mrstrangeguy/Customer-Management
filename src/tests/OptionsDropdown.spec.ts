import { it, expect, describe } from "vitest";
import { mount } from "@vue/test-utils";
import OptionsDropdown from "../components/OptionsDropdown.vue";

describe("OptionsDropdown", () => {
  const firstVariant = mount(OptionsDropdown, {
    props: {
      text: "This is header text",
      arrowIcon: "dx-icon dx-icon-spindown dx-icon-right",
    },
    slots: {
      "dropdown-items": `<div class="w-[94.97px] bg-[rgba(0,0,0,.12)] mt-1">
        <span
          class="block text-[13px] leading-[16px] px-[11px] pt-[10px] pb-[9px]"
          >All</span
        >
      </div>
      <div class="w-[94.97px] mt-1">
        <span
          class="block text-[13px] leading-[16px] px-[11px] pt-[10px] pb-[9px]"
          >All</span
        >
      </div>
      <div class="w-[94.97px] mt-1">
        <span
          class="block text-[13px] leading-[16px] px-[11px] pt-[10px] pb-[9px]"
          >All</span
        >
      </div>
      <div class="w-[94.97px] mt-1">
        <span
          class="block text-[13px] leading-[16px] px-[11px] pt-[10px] pb-[9px]"
          >All</span
        >
      </div>`,
    },
  });

  const secondVariant = mount(OptionsDropdown, {
    props: {
      icon: "dx-icon dx-icon-export",
      arrowIcon: "dx-icon dx-icon-spindown dx-icon-right",
    },
    slots: {
      "dropdown-items": `
     <div class="w-[94.97px] mt-1 flex w-[220.53px]">
        <div
          class="flex items-center text-[13px] leading-[16px] px-[11px] pt-[10px] pb-[9px]"
        >
          <div class="w-[29px]">
            <i
              class="default-icon font-medium text-[#0000008a] w-[18px] h-[18px] block dx-icon dx-icon-xlsxfile dx-list-item-icon text-[18px] leading-[18px]"
            ></i>
          </div>
          <div class="leading-[15px]">Export all data to Excel</div>
        </div>
      </div>
    `,
    },
  });

  
  const profileVariant = mount(OptionsDropdown, {
    props: {
      profileUrl:
        "https://js.devexpress.com/Demos/WidgetsGallery/JSDemos/images/employees/01.png",
    },
    slots: {
      "dropdown-items": `<div class="py-2.5 px-[11px] border-b border-[#e0e0e0]">
    <span class="block font-normal text-nowrap font-sm leading-[17.9998px] text-[14px] text-center">John Heart</span>
    </div>
    <div class="my-1">
    <div class="px-[11px] pt-2.5 pb-[9px] flex items-center">
    <div class="w-[29px] h-[18px]">
    <i class="default-icon dx-icon dx-icon-runner dx-list-item-icon leading-[18px] w-[18px] h-[18px] text-[18px] mr-4"/>
    </div>
    <span class="block leading-4 text-[13px]">Logout</span>
    </div>
    </div>`,
    },
  });

  it("check the snapshot variant 1", () => {
    expect(firstVariant.element).toMatchSnapshot();
  });

  it("check text prop", () => {
    expect(firstVariant.find("#heading-text").text()).toBe(
      "This is header text"
    );
  });

  it("check the snapshot variant 2", () => {
    expect(secondVariant.element).toMatchSnapshot();
  });


  it("check the snapshot of profile variant", () => {
    expect(profileVariant.element).toMatchSnapshot();
  });

  // it('check profile click event',() => {
  //   const header = firstVariant.element
  // })

});
