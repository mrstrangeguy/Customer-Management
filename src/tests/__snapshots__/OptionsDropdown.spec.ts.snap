// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OptionsDropdown > check the snapshot of profile variant 1`] = `
<div
  class="relative z-100 cursor-pointer w-fit"
>
  <!--v-if-->
  <img
    alt="profile-img"
    class="block w-[28px] h-[28px] object-cover object-top aspect-square border rounded-full"
    src="https://js.devexpress.com/Demos/WidgetsGallery/JSDemos/images/employees/01.png"
  />
  <div
    class="absolute cursor-pointer bg-white transition-all duration-[200ms] border-b w-auto p-px shadow-[0_2px_4px_rgba(0,0,0,.2)] opacity-0 z-[-1] right-0"
  >
    
    <div
      class="py-2.5 px-[11px] border-b border-[#e0e0e0]"
    >
      <span
        class="block font-normal text-nowrap font-sm leading-[17.9998px] text-[14px] text-center"
      >
        John Heart
      </span>
    </div>
    <div
      class="my-1"
    >
      <div
        class="px-[11px] pt-2.5 pb-[9px] flex items-center"
      >
        <div
          class="w-[29px] h-[18px]"
        >
          <i
            class="default-icon dx-icon dx-icon-runner dx-list-item-icon leading-[18px] w-[18px] h-[18px] text-[18px] mr-4"
          />
        </div>
        <span
          class="block leading-4 text-[13px]"
        >
          Logout
        </span>
      </div>
    </div>
    
  </div>
</div>
`;

exports[`OptionsDropdown > check the snapshot variant 1 1`] = `
<div
  class="relative z-100 cursor-pointer w-fit"
>
  <div
    class="relative overflow-hidden flex py-[5px] z-50 rounded-sm transition-all cursor-pointer pl-3 pr-2 w-fit hover:bg-[rgba(0,0,0,.08)]"
    role="button"
  >
    <!--v-if-->
    <span
      class="block text-[13px] leading-[18px] text-black font-medium tracking-[0.52px]"
      id="heading-text"
    >
      This is header text
    </span>
    <i
      class="default-icon ml-1 block h-[18px] w-4 text-base leading-4 text-center align-middle dx-icon dx-icon-spindown dx-icon-right"
    />
  </div>
  <!--v-if-->
  <div
    class="absolute cursor-pointer bg-white transition-all duration-[200ms] border-b w-auto p-px shadow-[0_2px_4px_rgba(0,0,0,.2)] opacity-0 z-[-1] top-[26px] left-0"
  >
    
    <div
      class="w-[94.97px] bg-[rgba(0,0,0,.12)] mt-1"
    >
      <span
        class="block text-[13px] leading-[16px] px-[11px] pt-[10px] pb-[9px]"
      >
        All
      </span>
    </div>
    <div
      class="w-[94.97px] mt-1"
    >
      <span
        class="block text-[13px] leading-[16px] px-[11px] pt-[10px] pb-[9px]"
      >
        All
      </span>
    </div>
    <div
      class="w-[94.97px] mt-1"
    >
      <span
        class="block text-[13px] leading-[16px] px-[11px] pt-[10px] pb-[9px]"
      >
        All
      </span>
    </div>
    <div
      class="w-[94.97px] mt-1"
    >
      <span
        class="block text-[13px] leading-[16px] px-[11px] pt-[10px] pb-[9px]"
      >
        All
      </span>
    </div>
    
  </div>
</div>
`;

exports[`OptionsDropdown > check the snapshot variant 2 1`] = `
<div
  class="relative z-100 cursor-pointer w-fit"
>
  <div
    class="relative overflow-hidden flex py-[5px] z-50 rounded-sm transition-all cursor-pointer pl-3 pr-2 w-fit hover:bg-[rgba(0,0,0,.08)]"
    role="button"
  >
    <i
      class="default-icon block font-normal block h-[18px] w-4 text-base leading-4 align-middle text-center dx-icon dx-icon-export"
    />
    <span
      class="block text-[13px] leading-[18px] text-black font-medium tracking-[0.52px]"
      id="heading-text"
    />
    <i
      class="default-icon ml-1 block h-[18px] w-4 text-base leading-4 text-center align-middle dx-icon dx-icon-spindown dx-icon-right"
    />
  </div>
  <!--v-if-->
  <div
    class="absolute cursor-pointer bg-white transition-all duration-[200ms] border-b w-auto p-px shadow-[0_2px_4px_rgba(0,0,0,.2)] opacity-0 z-[-1] top-[26px] left-0"
  >
    
    <div
      class="w-[94.97px] mt-1 flex w-[220.53px]"
    >
      <div
        class="flex items-center text-[13px] leading-[16px] px-[11px] pt-[10px] pb-[9px]"
      >
        <div
          class="w-[29px]"
        >
          <i
            class="default-icon font-medium text-[#0000008a] w-[18px] h-[18px] block dx-icon dx-icon-xlsxfile dx-list-item-icon text-[18px] leading-[18px]"
          />
        </div>
        <div
          class="leading-[15px]"
        >
          Export all data to Excel
        </div>
      </div>
    </div>
    
  </div>
</div>
`;
